drivers/digital-io/max22200/max22200.c

```diff
@@ -0,0 +1,910 @@
+	else
+		*val = no_os_get_unaligned_be32(desc->buff);
+
+	return 0;
+}
+
+/**
+ * @brief Write data to desired register for MAX22200
+ * @param desc - MAXX22200 device descriptor.
+ * @param reg - Register value to write data to.
+ * @param val - Value to be written to th register.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22200_reg_write(struct max22200_desc *desc, uint32_t reg, uint32_t val)
+{
+	int ret;
+	struct no_os_spi_msg xfer = {
+		.tx_buff = desc->buff,
+		.rx_buff = desc->buff,
+		.bytes_number = 1,
+		.cs_change = 1,
+	};
+
+	if (no_os_field_get(reg, MAX22200_R1B) && val > 0xFF)
+		return -EINVAL;
+
+	desc->buff[0] = no_os_field_prep(MAX22200_RW_MASK, 1) |
+			no_os_field_prep(MAX22200_ADDR_MASK, reg);
+
+	ret = no_os_gpio_set_value(desc->cmd_desc, NO_OS_GPIO_HIGH);
+	if (ret)
+		return ret;
+
+	ret = no_os_spi_transfer(desc->comm_desc, &xfer, 1);
+	if (ret)
+		return ret;
+
+	if (!no_os_field_get(reg, MAX22200_R1B)) {
+		xfer.bytes_number = MAX22200_FRAME_SIZE;
+		no_os_put_unaligned_le32(val, desc->buff);
+	} else {
+		desc->buff[0] = val;
+	}
+
+	ret = no_os_gpio_set_value(desc->cmd_desc, NO_OS_GPIO_LOW);
+	if (ret)
+		return ret;
+
+	return no_os_spi_transfer(desc->comm_desc, &xfer, 1);
+}
+
+/**
+ * @brief Update data in the desired register.
+ * @param desc - MAX22200 device descriptor.
+ * @param reg - Register value to update data to.
+ * @param mask - Bit mask of the field to be updated.
+ * @param val - Value to be updated in the desired register.
+ * 		Should be bit shifted by using no_os_field_prep(mask, val)
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22200_reg_update(struct max22200_desc *desc, uint32_t reg, uint32_t mask,
+			uint32_t val)
+{
+	int ret;
+	uint32_t reg_val = 0;
+
+	ret = max22200_reg_read(desc, reg, &reg_val);
+	if (ret)
+		return ret;
+
+	reg_val &= ~mask;
+	reg_val |= mask & val;
+
+	return max22200_reg_write(desc, reg, reg_val);
+}
+
+/**
+ * @brief Set external trigger's state of the MAX22200
+ * @param desc - MAX22200 device descriptor.
+ * @param trig_state - State to change to.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22200_set_trig_state(struct max22200_desc *desc,
+			    enum max22200_trig_state trig_state)
+{
+	if (!desc->trig_desc)
+		return -EINVAL;
+
+	switch (trig_state) {
+	case MAX22200_TRIG_OFF:
+		return no_os_gpio_set_value(desc->trig_desc, NO_OS_GPIO_LOW);
+	case MAX22200_TRIG_ON:
+		return no_os_gpio_set_value(desc->trig_desc, NO_OS_GPIO_HIGH);
+	default:
+		return -EINVAL;
+	}
+}
+
+/**
+ * @brief Set channel state for specific channel
+ * @param desc - MAX22200 device descriptor.
+ * @param ch - Channel number.
+ * @param ch_state - Channel's state to be changed into.
+ * @return 0 in case of succces, negative error code otherwise.
+*/
+int max22200_set_ch_state(struct max22200_desc *desc, uint32_t ch,
+			  enum max22200_ch_state ch_state)
+{
+	if (ch > MAX22200_CHANNELS - 1)
```

Comment:
I would rename this to MAX22200_MAX_CHN_IDX and give it a value of 7, and make the updates everywhere.

Reply:
will do.